The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7254937816/job/19764619804) [(plain)](https://github.com/rust-lang/rust/commit/6c381049ad72221edc02a089ac11e6a26703f401/checks/19764619804/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Built container sha256:9c3c93a371e5aed5c18185b24f130d95d5140dbd72a9b325e7b6b49e521a4faa
Looks like docker image is the same as before, not uploading
https://ci-caches.rust-lang.org/docker/7ebc15c01a233894034d277c8cce4e949f4e7791f66b4727c8fb6e058a0b8171d6152e1441d677cef0653843ceeee469c097b8699b2bb74249e674f6aa1a8813
sha256:9c3c93a371e5aed5c18185b24f130d95d5140dbd72a9b325e7b6b49e521a4faa
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-16]
##[group]Clock drift check
  local time: Mon Dec 18 23:12:42 UTC 2023
  network time: Mon, 18 Dec 2023 23:12:42 GMT
  network time: Mon, 18 Dec 2023 23:12:42 GMT
##[endgroup]
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-16', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-missing-tools', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-16/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: rust.codegen-units-std := 1
---
........................................................................................  7480/15965
............i...........................................................................  7568/15965
....................................................................i...................  7656/15965
........................................................................................  7744/15965
.................i.........................................................ii.ii...F....  7832/15965
....................i...................................................................  8008/15965
........................................................................................  8096/15965
................................i....i..................................................  8184/15965
............................i................i..........................................  8272/15965
---
........................................................................................ 14784/15965
........................................................................................ 14872/15965
........................................................................................ 14960/15965
........................................................................................ 15048/15965
FF....F................................................................................. 15136/15965
........................................................................................ 15312/15965
........................................................................................ 15400/15965
........................................................................................ 15488/15965
........................................................................................ 15576/15965
---

---- [ui] tests/ui/issues/issue-22638.rs stdout ----
diff of stderr:

1 error: reached the recursion limit while instantiating `A::matches::$CLOSURE`
+   --> $DIR/issue-22638.rs:57:9
3    |
3    |
4 LL |         a.matches(f)

6    |
6    |
7 note: `A::matches` defined here
+   --> $DIR/issue-22638.rs:16:5
9    |
9    |
10 LL |     pub fn matches<F: Fn()>(&self, f: &F) {


The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-22638/issue-22638.stderr
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-22638/issue-22638.stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args issues/issue-22638.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/issues/issue-22638.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-22638" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-22638/auxiliary"
--- stderr -------------------------------
--- stderr -------------------------------
error: reached the recursion limit while instantiating `A::matches::<{closure@/checkout/tests/ui/issues/issue-22638.rs:45:23: 45:25}>`
   |
   |
LL |         a.matches(f)
   |
   |
note: `A::matches` defined here
   |
   |
LL |     pub fn matches<F: Fn()>(&self, f: &F) {

error: aborting due to 1 previous error
------------------------------------------



---- [ui] tests/ui/issues/issue-67552.rs stdout ----
diff of stderr:

1 error: reached the recursion limit while instantiating `rec::<&mut &mut &mut &mut &mut ...>`
+   --> $DIR/issue-67552.rs:29:9
3    |
3    |
4 LL |         rec(identity(&mut it))

6    |
7 note: `rec` defined here
-   --> $DIR/issue-67552.rs:23:1
---
To only update this specific test, also pass `--test-args issues/issue-67552.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/issues/issue-67552.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-67552" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-67552/auxiliary" "-Copt-level=0"
--- stderr -------------------------------
--- stderr -------------------------------
error: reached the recursion limit while instantiating `rec::<&mut &mut &mut &mut &mut ...>`
   |
   |
LL |         rec(identity(&mut it))
   |
note: `rec` defined here
  --> /checkout/tests/ui/issues/issue-67552.rs:22:1
   |
   |
LL | / fn rec<T>(mut it: T)
LL | | where
LL | |     T: Iterator,
   | |________________^
   = note: the full type name has been written to '/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-67552/issue-67552.long-type.txt'
error: aborting due to 1 previous error
------------------------------------------



---- [ui] tests/ui/type-alias-impl-trait/self-referential.rs stdout ----
diff of stderr:

1 error[E0277]: can't compare `&i32` with `Bar<'b, 'a>`
+   --> $DIR/self-referential.rs:7:31
3    |
3    |
4 LL | fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
5    |                               ^^^^^^^^^^^ no implementation for `&i32 == Bar<'b, 'a>`
11    = help: the trait `PartialEq` is implemented for `i32`
12 
12 
13 error[E0277]: can't compare `&i32` with `(i32, Foo<'a, 'b>::{opaque#0})`
+   --> $DIR/self-referential.rs:14:31
15    |
15    |
16 LL | fn foo<'a, 'b>(i: &'a i32) -> Foo<'a, 'b> {
17    |                               ^^^^^^^^^^^ no implementation for `&i32 == (i32, Foo<'a, 'b>::{opaque#0})`
23    = help: the trait `PartialEq` is implemented for `i32`
24 
24 
25 error[E0277]: can't compare `&i32` with `(i32, Moo<'b, 'a>::{opaque#0})`
+   --> $DIR/self-referential.rs:21:31
27    |
27    |
28 LL | fn moo<'a, 'b>(i: &'a i32) -> Moo<'a, 'b> {
29    |                               ^^^^^^^^^^^ no implementation for `&i32 == (i32, Moo<'b, 'a>::{opaque#0})`

The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type-alias-impl-trait/self-referential/self-referential.stderr
To update references, rerun the tests and pass the `--bless` flag
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args type-alias-impl-trait/self-referential.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/type-alias-impl-trait/self-referential.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type-alias-impl-trait/self-referential" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type-alias-impl-trait/self-referential/auxiliary"
stdout: none
--- stderr -------------------------------
error[E0277]: can't compare `&i32` with `Bar<'b, 'a>`
   |
   |
LL | fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
   |                               ^^^^^^^^^^^ no implementation for `&i32 == Bar<'b, 'a>`
LL |     //~^ ERROR can't compare `&i32` with `Bar<'b, 'a>`
LL |     i
   |     - return type was inferred to be `&i32` here
   |
   = help: the trait `PartialEq<Bar<'b, 'a>>` is not implemented for `&i32`
   = help: the trait `PartialEq` is implemented for `i32`

error[E0277]: can't compare `&i32` with `(i32, Foo<'a, 'b>::{opaque#0})`
   |
   |
LL | fn foo<'a, 'b>(i: &'a i32) -> Foo<'a, 'b> {
   |                               ^^^^^^^^^^^ no implementation for `&i32 == (i32, Foo<'a, 'b>::{opaque#0})`
LL |     //~^ ERROR can't compare `&i32` with `(i32, Foo<'a, 'b>::{opaque#0})`
LL |     (42, i)
   |     ------- return type was inferred to be `(i32, &i32)` here
   |
   = help: the trait `PartialEq<(i32, Foo<'a, 'b>::{opaque#0})>` is not implemented for `&i32`
   = help: the trait `PartialEq` is implemented for `i32`

error[E0277]: can't compare `&i32` with `(i32, Moo<'b, 'a>::{opaque#0})`
   |
   |
LL | fn moo<'a, 'b>(i: &'a i32) -> Moo<'a, 'b> {
   |                               ^^^^^^^^^^^ no implementation for `&i32 == (i32, Moo<'b, 'a>::{opaque#0})`
LL |     //~^ ERROR can't compare `&i32` with `(i32, Moo<'b, 'a>::{opaque#0})`
LL |     (42, i)
   |     ------- return type was inferred to be `(i32, &i32)` here
   |
   = help: the trait `PartialEq<(i32, Moo<'b, 'a>::{opaque#0})>` is not implemented for `&i32`
   = help: the trait `PartialEq` is implemented for `i32`
error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0277`.
------------------------------------------
------------------------------------------


---- [ui] tests/ui/type-alias-impl-trait/self-referential-4.rs stdout ----
diff of stderr:

1 error[E0277]: can't compare `&i32` with `Bar<'b, 'static>`
+   --> $DIR/self-referential-4.rs:7:31
3    |
3    |
4 LL | fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
5    |                               ^^^^^^^^^^^ no implementation for `&i32 == Bar<'b, 'static>`
10    = help: the trait `PartialEq` is implemented for `i32`
11 
11 
12 error[E0277]: can't compare `&i32` with `Foo<'static, 'b>`
+   --> $DIR/self-referential-4.rs:13:31
14    |
14    |
15 LL | fn foo<'a, 'b>(i: &'a i32) -> Foo<'a, 'b> {
16    |                               ^^^^^^^^^^^ no implementation for `&i32 == Foo<'static, 'b>`
21    = help: the trait `PartialEq` is implemented for `i32`
22 
22 
23 error[E0277]: can't compare `&i32` with `Moo<'static, 'a>`
+   --> $DIR/self-referential-4.rs:19:31
25    |
25    |
26 LL | fn moo<'a, 'b>(i: &'a i32) -> Moo<'a, 'b> {
27    |                               ^^^^^^^^^^^ no implementation for `&i32 == Moo<'static, 'a>`

The actual stderr differed from the expected stderr.
The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type-alias-impl-trait/self-referential-4/self-referential-4.stderr
To only update this specific test, also pass `--test-args type-alias-impl-trait/self-referential-4.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/type-alias-impl-trait/self-referential-4.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type-alias-impl-trait/self-referential-4" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type-alias-impl-trait/self-referential-4/auxiliary"
--- stderr -------------------------------
--- stderr -------------------------------
error[E0277]: can't compare `&i32` with `Bar<'b, 'static>`
   |
   |
LL | fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
   |                               ^^^^^^^^^^^ no implementation for `&i32 == Bar<'b, 'static>`
LL |     i //~^ ERROR can't compare `&i32` with `Bar<'b, 'static>`
   |     - return type was inferred to be `&i32` here
   |
   = help: the trait `PartialEq<Bar<'b, 'static>>` is not implemented for `&i32`
   = help: the trait `PartialEq` is implemented for `i32`

error[E0277]: can't compare `&i32` with `Foo<'static, 'b>`
   |
   |
LL | fn foo<'a, 'b>(i: &'a i32) -> Foo<'a, 'b> {
   |                               ^^^^^^^^^^^ no implementation for `&i32 == Foo<'static, 'b>`
LL |     i //~^ ERROR can't compare `&i32` with `Foo<'static, 'b>`
   |     - return type was inferred to be `&i32` here
   |
   = help: the trait `PartialEq<Foo<'static, 'b>>` is not implemented for `&i32`
   = help: the trait `PartialEq` is implemented for `i32`

error[E0277]: can't compare `&i32` with `Moo<'static, 'a>`
   |
   |
LL | fn moo<'a, 'b>(i: &'a i32) -> Moo<'a, 'b> {
   |                               ^^^^^^^^^^^ no implementation for `&i32 == Moo<'static, 'a>`
LL |     i //~^ ERROR can't compare `&i32` with `Moo<'static, 'a>`
   |     - return type was inferred to be `&i32` here
   |
   = help: the trait `PartialEq<Moo<'static, 'a>>` is not implemented for `&i32`
   = help: the trait `PartialEq` is implemented for `i32`
error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0277`.
------------------------------------------
------------------------------------------


---- [ui] tests/ui/type-alias-impl-trait/self-referential-3.rs stdout ----
diff of stderr:

1 error[E0277]: can't compare `&i32` with `Bar<'a, 'b>`
+   --> $DIR/self-referential-3.rs:7:31
3    |
3    |
4 LL | fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
5    |                               ^^^^^^^^^^^ no implementation for `&i32 == Bar<'a, 'b>`

The actual stderr differed from the expected stderr.
The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type-alias-impl-trait/self-referential-3/self-referential-3.stderr
To only update this specific test, also pass `--test-args type-alias-impl-trait/self-referential-3.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/type-alias-impl-trait/self-referential-3.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type-alias-impl-trait/self-referential-3" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type-alias-impl-trait/self-referential-3/auxiliary"
--- stderr -------------------------------
--- stderr -------------------------------
error[E0277]: can't compare `&i32` with `Bar<'a, 'b>`
   |
   |
LL | fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
   |                               ^^^^^^^^^^^ no implementation for `&i32 == Bar<'a, 'b>`
LL |     //~^ ERROR can't compare `&i32` with `Bar<'a, 'b>`
LL |     i
   |     - return type was inferred to be `&i32` here
   |
   = help: the trait `PartialEq<Bar<'a, 'b>>` is not implemented for `&i32`
   = help: the trait `PartialEq` is implemented for `i32`
error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
------------------------------------------
```

</details>

--- a/tests/ui/issues/issue-22638.rs
+++ b/tests/ui/issues/issue-22638.rs
@@ -1,6 +1,7 @@
 // build-fail
 // normalize-stderr-test: "<\{closure@.+`" -> "$$CLOSURE`"
 // normalize-stderr-test: ".nll/" -> "/"
+// ignore-compare-mode-next-solver (hangs)
 
 #![allow(unused)]
 

--- a/tests/ui/issues/issue-22638.stderr
+++ b/tests/ui/issues/issue-22638.stderr
@@ -1,11 +1,11 @@
 error: reached the recursion limit while instantiating `A::matches::$CLOSURE`
-  --> $DIR/issue-22638.rs:56:9
+  --> $DIR/issue-22638.rs:57:9
    |
 LL |         a.matches(f)
    |         ^^^^^^^^^^^^
    |
 note: `A::matches` defined here
-  --> $DIR/issue-22638.rs:15:5
+  --> $DIR/issue-22638.rs:16:5
    |
 LL |     pub fn matches<F: Fn()>(&self, f: &F) {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

--- a/tests/ui/issues/issue-67552.rs
+++ b/tests/ui/issues/issue-67552.rs
@@ -1,7 +1,6 @@
 // build-fail
 // compile-flags: -Copt-level=0
 // normalize-stderr-test: ".nll/" -> "/"
-// ignore-compare-mode-next-solver (hangs)
 
 fn main() {
     rec(Empty);

--- a/tests/ui/issues/issue-67552.stderr
+++ b/tests/ui/issues/issue-67552.stderr
@@ -1,11 +1,11 @@
 error: reached the recursion limit while instantiating `rec::<&mut &mut &mut &mut &mut ...>`
-  --> $DIR/issue-67552.rs:30:9
+  --> $DIR/issue-67552.rs:29:9
    |
 LL |         rec(identity(&mut it))
    |         ^^^^^^^^^^^^^^^^^^^^^^
    |
 note: `rec` defined here
-  --> $DIR/issue-67552.rs:23:1
+  --> $DIR/issue-67552.rs:22:1
    |
 LL | / fn rec<T>(mut it: T)
 LL | | where

--- a/tests/ui/recursion/issue-95134.rs
+++ b/tests/ui/recursion/issue-95134.rs
@@ -3,7 +3,6 @@
 // compile-flags: -Copt-level=0
 // dont-check-failure-status
 // dont-check-compiler-stderr
-// ignore-compare-mode-next-solver (hangs)
 
 pub fn encode_num<Writer: ExampleWriter>(n: u32, mut writer: Writer) -> Result<(), Writer::Error> {
     if n > 15 {

--- a/tests/ui/type-alias-impl-trait/self-referential-3.rs
+++ b/tests/ui/type-alias-impl-trait/self-referential-3.rs
@@ -1,3 +1,5 @@
+// ignore-compare-mode-next-solver (hangs)
+
 #![feature(type_alias_impl_trait)]
 
 type Bar<'a, 'b> = impl PartialEq<Bar<'a, 'b>> + std::fmt::Debug;

--- a/tests/ui/type-alias-impl-trait/self-referential-3.stderr
+++ b/tests/ui/type-alias-impl-trait/self-referential-3.stderr
@@ -1,5 +1,5 @@
 error[E0277]: can't compare `&i32` with `Bar<'a, 'b>`
-  --> $DIR/self-referential-3.rs:5:31
+  --> $DIR/self-referential-3.rs:7:31
    |
 LL | fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
    |                               ^^^^^^^^^^^ no implementation for `&i32 == Bar<'a, 'b>`

--- a/tests/ui/type-alias-impl-trait/self-referential-4.rs
+++ b/tests/ui/type-alias-impl-trait/self-referential-4.rs
@@ -1,3 +1,5 @@
+// ignore-compare-mode-next-solver (hangs)
+
 #![feature(type_alias_impl_trait)]
 
 type Bar<'a, 'b> = impl PartialEq<Bar<'b, 'static>> + std::fmt::Debug;

--- a/tests/ui/type-alias-impl-trait/self-referential-4.stderr
+++ b/tests/ui/type-alias-impl-trait/self-referential-4.stderr
@@ -1,5 +1,5 @@
 error[E0277]: can't compare `&i32` with `Bar<'b, 'static>`
-  --> $DIR/self-referential-4.rs:5:31
+  --> $DIR/self-referential-4.rs:7:31
    |
 LL | fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
    |                               ^^^^^^^^^^^ no implementation for `&i32 == Bar<'b, 'static>`
@@ -10,7 +10,7 @@ LL |     i
    = help: the trait `PartialEq` is implemented for `i32`
 
 error[E0277]: can't compare `&i32` with `Foo<'static, 'b>`
-  --> $DIR/self-referential-4.rs:11:31
+  --> $DIR/self-referential-4.rs:13:31
    |
 LL | fn foo<'a, 'b>(i: &'a i32) -> Foo<'a, 'b> {
    |                               ^^^^^^^^^^^ no implementation for `&i32 == Foo<'static, 'b>`
@@ -21,7 +21,7 @@ LL |     i
    = help: the trait `PartialEq` is implemented for `i32`
 
 error[E0277]: can't compare `&i32` with `Moo<'static, 'a>`
-  --> $DIR/self-referential-4.rs:17:31
+  --> $DIR/self-referential-4.rs:19:31
    |
 LL | fn moo<'a, 'b>(i: &'a i32) -> Moo<'a, 'b> {
    |                               ^^^^^^^^^^^ no implementation for `&i32 == Moo<'static, 'a>`

--- a/tests/ui/type-alias-impl-trait/self-referential.rs
+++ b/tests/ui/type-alias-impl-trait/self-referential.rs
@@ -1,3 +1,5 @@
+// ignore-compare-mode-next-solver (hangs)
+
 #![feature(type_alias_impl_trait)]
 
 type Bar<'a, 'b> = impl PartialEq<Bar<'b, 'a>> + std::fmt::Debug;

--- a/tests/ui/type-alias-impl-trait/self-referential.stderr
+++ b/tests/ui/type-alias-impl-trait/self-referential.stderr
@@ -1,5 +1,5 @@
 error[E0277]: can't compare `&i32` with `Bar<'b, 'a>`
-  --> $DIR/self-referential.rs:5:31
+  --> $DIR/self-referential.rs:7:31
    |
 LL | fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
    |                               ^^^^^^^^^^^ no implementation for `&i32 == Bar<'b, 'a>`
@@ -11,7 +11,7 @@ LL |     i
    = help: the trait `PartialEq` is implemented for `i32`
 
 error[E0277]: can't compare `&i32` with `(i32, Foo<'a, 'b>::{opaque#0})`
-  --> $DIR/self-referential.rs:12:31
+  --> $DIR/self-referential.rs:14:31
    |
 LL | fn foo<'a, 'b>(i: &'a i32) -> Foo<'a, 'b> {
    |                               ^^^^^^^^^^^ no implementation for `&i32 == (i32, Foo<'a, 'b>::{opaque#0})`
@@ -23,7 +23,7 @@ LL |     (42, i)
    = help: the trait `PartialEq` is implemented for `i32`
 
 error[E0277]: can't compare `&i32` with `(i32, Moo<'b, 'a>::{opaque#0})`
-  --> $DIR/self-referential.rs:19:31
+  --> $DIR/self-referential.rs:21:31
    |
 LL | fn moo<'a, 'b>(i: &'a i32) -> Moo<'a, 'b> {
    |                               ^^^^^^^^^^^ no implementation for `&i32 == (i32, Moo<'b, 'a>::{opaque#0})`

